src/service/mpris.ts

```diff
@@ -163,10 +202,46 @@ class MprisPlayer extends GObject.Object {
                     this.emit('changed');
                 }
                 catch (e) {
-                    logError(e as Error, `failed to cache ${coverPath}`);
+                    logError(e as Error, `failed to cache ${trackCoverUrl}`);
+                    return "";
```

Comment:
this is an async function this return value wont get assigned to coverPath
and even if there was an error caching, the function continues to mark it cached in the json file

Reply:
done